django/src/rdwatch/tasks/__init__.py

```diff
@@ -1,6 +1,275 @@
+    matchConstellation = ''
+    baseSiteEval = None
+    # First we gather all images that match observations
+    for observation in site_observations:
+        min_time = min(min_time, observation.timestamp)
+        max_time = max(max_time, observation.timestamp)
+        mercator: tuple[float, float, float, float] = observation.geom.extent
+        tempbox = transformer.transform_bounds(
+            mercator[0], mercator[1], mercator[2], mercator[3]
+        )
+        bbox = [tempbox[1], tempbox[0], tempbox[3], tempbox[2]]
+        bbox = scale_bbox(bbox, 1.2)
+        max_bbox = get_max_bbox(bbox, max_bbox)
+        timestamp = observation.timestamp
+        constellation = observation.constellation
+        # We need to grab the image for this timerange and type
+        if str(constellation) == baseConstellation:
+            baseSiteEval = observation.siteeval
+            matchConstellation = constellation
+            found = SiteImage.objects.filter(
+                siteeval=observation.siteeval,
+                siteobs=observation,
+                timestamp=observation.timestamp,
+                source=baseConstellation,
+            )
+            if found.count() > 0 and not force:
+                found_timestamps[observation.timestamp] = True
+                continue
+            results = fetch_boundbox_image(
+                bbox, timestamp, constellation, baseConstellation == 'WV'
+            )
+            if results is None:
+                logger.warning(f'COULD NOT FIND ANY IMAGE FOR TIMESTAMP: {timestamp}')
+                continue
+            bytes = results['bytes']
+            cloudcover = results['cloudcover']
+            found_timestamp = results['timestamp']
+            if bytes is None:
+                logger.warning(f'COULD NOT FIND ANY IMAGE FOR TIMESTAMP: {timestamp}')
+                continue
+            found_timestamps[found_timestamp] = True
+            # logger.warning(f'Retrieved Image with timestamp: {timestamp}')
+            output = f'tile_image_{observation.id}.jpg'
+            with open(output, 'wb') as f:
+                f.write(bytes)
+            with open(output, 'rb') as imageFile:
+                image = File(imageFile, output)
+                if found.count() > 0:
+                    existing = found.first()
+                    existing.image.delete()  # remove previous image if new one found
+                    existing.cloudcover = cloudcover
+                    existing.image = image
+                    existing.save()
+                else:
+                    SiteImage.objects.create(
+                        siteeval=observation.siteeval,
+                        siteobs=observation,
+                        timestamp=observation.timestamp,
+                        image=image,
+                        cloudcover=cloudcover,
+                        source=baseConstellation,
+                    )
+            os.remove(output)
```

Comment:
Can we use a [`TemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile) here instead of managing the file's lifecycle ourselves?